src/devices/localization2DServer/Localization2DServer.cpp

```diff
@@ -39,14 +40,23 @@ using namespace yarp::dev::Nav2D;
 using namespace std;
 
 #define DEFAULT_THREAD_PERIOD 0.01
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
 //------------------------------------------------------------------------------------------------------------------------------
 
 Localization2DServer::Localization2DServer() : PeriodicThread(DEFAULT_THREAD_PERIOD)
 {
+    m_ros_node = nullptr;
+    m_current_status = yarp::dev::Nav2D::LocalizationStatusEnum::localization_status_not_yet_localized;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'm_current_status' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=4964328)